Source/Core/VideoBackends/OGL/VertexManager.cpp

```diff
@@ -140,6 +140,15 @@ void VertexManager::vFlush(bool useDstAlpha)
 
 	PrepareDrawBuffers(stride);
 
+	if (!bpmem.genMode.zfreeze && IndexGenerator::GetIndexLen() >= 3)
+	{
+		CalculateZSlope(stride);
```

Comment:
You can do this like all the other shader managers do, make a callback in PixelShaderManager which is called from BPStructs.cpp when the zfreeze changes. Then you can make sure you only mark the uniform struct dirty when the zfreeze mode is actually changed.